### PR TITLE
(fleet) embed systemd units in the installer binary

### PR DESCRIPTION
**Pull request by arbll**
This PR emebeds the agent and installer systemd units in the binary. This will make the "download installer every time" easier to implement.

### CODE REVIEW

**Review by BaptisteFoy — APPROVED** [review]

File: pkg/fleet/installer/service/embedded/datadog-installer-exp.service [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+OnFailure=datadog-installer.service
+Conflicts=datadog-installer.service
+JobTimeoutSec=3000 #50 minutes
+
+[Service]
+Type=oneshot
+PIDFile=/var/run/datadog/installer/installer.pid
+ExecStart=/opt/datadog-packages/datadog-installer/stable/bin/installer/installer run -p /var/run/datadog/installer/installer.pid
+ExecStart=/opt/datadog-packages/datadog-installer/stable/bin/installer/installer run -p /var/run/datadog/installer/installer.pid
+ExecStart=/opt/datadog-packages/datadog-installer/stable/bin/installer/installer run -p /var/run/datadog/installer/installer.pid
```
Comment (chouquette) [inline pkg/fleet/installer/service/embedded/datadog-installer-exp.service:13]: Should't the path be `/opt/..../experiment/bin/...` here?

Comment (arbll) [reply pkg/fleet/installer/service/embedded/datadog-installer-exp.service]: good catch

**Review by chouquette — APPROVED** [review]

File: pkg/fleet/installer/service/embedded/datadog-agent-trace-exp.service [hunk 2]
```diff
@@ -4,11 +4,11 @@ BindsTo=datadog-agent-exp.service
 
 [Service]
 Type=simple
-PIDFile=<%= agent_dir %>/run/trace-agent.pid
+PIDFile=/opt/datadog-packages/datadog-agent/experiment/run/trace-agent.pid
 User=dd-agent
 Restart=on-failure
 EnvironmentFile=-<%= etc_dir %>/environment
-ExecStart=<%= agent_dir %>/embedded/bin/trace-agent --config <%= etc_dir %>/datadog.yaml --pidfile <%= agent_dir %>/run/trace-agent.pid
+ExecStart=/opt/datadog-packages/datadog-agent/experiment/embedded/bin/trace-agent --config <%= etc_dir %>/datadog.yaml --pidfile /opt/datadog-packages/datadog-agent/experiment/run/trace-agent.pid
```
Comment (raphaelgavache) [inline pkg/fleet/installer/service/embedded/datadog-agent-trace-exp.service:11]: there are some <%= etc_dir %>

File: pkg/fleet/installer/service/embedded/datadog-installer-exp.service [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=Datadog Installer Experiment
+After=network.target
+OnFailure=datadog-installer.service
+Conflicts=datadog-installer.service
+JobTimeoutSec=3000 #50 minutes
+
+[Service]
+Type=oneshot
+PIDFile=/var/run/datadog/installer/installer.pid
```
Comment (raphaelgavache) [inline pkg/fleet/installer/service/embedded/datadog-installer-exp.service:10]: datadog/installer -> datadog-installer

**Pull request merged**